packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -5355,8 +5361,28 @@ abstract class RenderObjectElement extends Element {
     widget.didUnmountRenderObject(renderObject);
   }
 
-  void _updateParentData(ParentDataWidget<RenderObjectWidget> parentData) {
-    parentData.applyParentData(renderObject);
+  void _updateParentData(ParentDataWidget<ParentData> parentDataWidget) {
+    bool applyParentData = true;
+    assert(() {
+      try {
```

Comment:
Ditto here.  We should avoid using exceptions for flow control - it causes debuggers to fire here when users are debugging their own code.

Or is that whta we want?  If so, we should add a comment to that effect.

Reply:
This is not an exception for control flow. This throws when there's actually something wrong with your app, so if you are debugging your app and you have it set up to fire on exception, you want it to fire here.

I am only catching the exception here directly, because I don't want the exception to bubble up and cause the error widget to swoop in because the tree is in a state where the error widget would cause more exceptions...

Reply:
> Maybe just q comment then about why we are throwing and catching this way would help

Will do.